### ISSUE TITLE
no-return-await false positive

### ISSUE THREAD

**Issue by ljw1004** [eslint/eslint#8255]
Version: ESLint v3.17.1

```js
async function foo() {
  await baz();
  return await bar();
}
```

The no-return-await rule files in this case. But it shouldn't: there's no way to remove the "return await bar" and yet keep the "await baz".

In implementation terms, currently it does this:
```js
if (node.parent.type === "ReturnStatement") {
  return !hasErrorHandler(node.parent);
}
```

I think it should do this:
```js
if (node.parent.type === "ReturnStatement") {
  return !hasErrorHandler(node.parent) && !bodyHasOtherAwaits(node.parent);
}
```

**Comment by not-an-aardvark** [comment]
This is working as intended. You can replace your code with

```js
async function foo() {
  await baz();
  return bar();
}
```

**Comment by ljw1004** [comment]
Okay, I guess I'd misunderstood how ES promises work. I guess they automatically fold promises. (In C# if you tried the same return value then you'd end up returning a promise of a promise).

**Comment by not-an-aardvark** [comment]
That's right -- in JS Promises always fold, so you can't ever have a promise of a promise.
